### DISCUSSION TITLE
What’s the range of one block?

### DISCUSSION THREAD

**Discussion by nobiot** [Kinneyzhang/roam-block#13] [Q&A] [answered]
Hi @Kinneyzhang,
Thank you. I love the way embed is done:

> For block-embed, you can change the content of any embed blocks and all the blocks which are derived from one original block will be synchronized. 

For embed, I don’t seem to be able to use a marked region, but only “at-point”. What range of text does embed take? A paragraph? 

I will look deeper in the code, but wondered if you have a specific design in mind. Ref seems to work for a selected region.

**Comment by Kinneyzhang** [comment, marked as answer]
Thanks for you feedback @nobiot !

The concept of block in block-ref and block-embed is the same. In my mind, a block should be a minimum range of text shared one element type.

The reason why users cannot store many embed blocks is that I forget to implement it. But it should not be difficult to implement. Thanks for your idea！

BTW, I have got some new ideas: https://github.com/Kinneyzhang/roam-block/discussions/14#discussion-1153950. Maybe you could give me some suggestions. :-)

**Reply by nobiot** [reply]
Thank you. Looking forward to the enhancement. I will look at the other idea, too. Thanks!

**Comment by 101scholar** [comment]
Technically, the concept of block depends on what type of file you are in. In org file, each block is a paragraph. You can see what current block is by `(org-element-at-point)`. For other file types, it has not been implemented yet.
